lib/model-builder.js

```diff
@@ -168,6 +169,9 @@ ModelBuilder.prototype.define = function defineClass(className, properties, sett
     return ModelBaseClass.extend(className, properties, settings);
   }
 
+  // initializing base model inheritance rank if not set already
+  ModelBaseClass.__rank = ModelBaseClass.__rank || 1;
```

Comment:
It makes me wonder: is is possible to compute this property dynamically by crawling the inheritance chain? The following code seems to work well for me (Node.js versions 4 and 6):

```js
function computeInheritanceDepth(ctor, rootCtor) {
  if (!rootCtor) rootCtor = Object;
  let depth = 0;
  while (ctor && ctor !== rootCtor) {
    depth++;
    ctor = ctor.prototype &&
      ctor.prototype.__proto__ &&
      ctor.prototype.__proto__.constructor;
  }
  return depth;
}

// usage:
const rank = computeInheritanceDepth(ModelClass, DefaultModelBaseClass)
// => 0 for ModelBaseClass
// => 1 for User based on ModelBaseClass
// => 2 for Customer based on User
// etc.
```